pipeline/scripts/ci/upstream_cli.py

```diff
@@ -0,0 +1,174 @@
+"""
+This script provides the upstream latest rpm repo path and container image.
+Please note that this script should be run under root privilages
+inorder to pull the image and podman should be installed on that host.
+"""
+import fcntl
+import logging
+import os
+import re
+import sys
+import time
+
+import requests
+import yaml
+from docopt import docopt
+from packaging import version
+
+LOG = logging.getLogger(__name__)
+
+
+def store_in_upstream(branch_name, repo, image, upstream_version):
+    """Method to update it in upstream.yaml.
+    Args:
+        branch_name: Name of upstream branch
+        repo: upstream branch compose URL
+        image: upstream branch image
+        version: upstream branch ceph version.
+    """
+    file_name = "/ceph/cephci-jenkins/latest-rhceph-container-info/upstream.yaml"
+    stream = open(file_name, "r")
+    repo_details = yaml.load(stream)
+    # To handle new release which is not updated yet
+    if repo_details[branch_name]:
+        existing_version = repo_details[branch_name]["ceph-version"]
+    else:
+        # updating new branch details in upstream.yaml
+        repo_details[branch_name] = {
+            "ceph-version": upstream_version,
+            "composes": repo,
+            "image": image,
+        }
+        with open(file_name, "w") as yaml_file:
```

Comment:
needs locking mechanism here as well.

Reply:
can we make it common call, once you have all details to be updated.

Reply:
@sunilkumarn417 It happens only at the first time when new upstream branch is released, So in real time it won't deal with race conditions among jobs, Hence I feel lock not needed there.

Reply:
@sunilkumarn417 yes handled locking there as well and made it file update as a common call 
Tested both below scenario
1) existing branch update successfully
2) new branch also gets updated in the same format successfully.
Thanks for the review, Please review it also tested both